micrometer-core/src/main/java/io/micrometer/core/instrument/binder/logging/LogbackMetrics.java

```diff
@@ -51,6 +51,12 @@ public class LogbackMetrics implements MeterBinder, AutoCloseable {
 
     private final Map<MeterRegistry, MetricsTurboFilter> metricsTurboFilters = new HashMap<>();
 
+    static {
+        // see gh-2868. Without this called statically, the same call in the constructor
+        // may return SubstituteLoggerFactory and fail to cast.
+        LoggerFactory.getILoggerFactory();
```

Comment:
This feels hacky and I'm not sure if it is a terrible idea for some reason I couldn't think of yet. The issue we're fixing largely feels like one that is probably limited to running tests. For that reason I'm not sure it is worth doing this or not, but it does seem to resolve the issue at least in our tests.